pandas/tests/extension/test_stack.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+
+import pandas as pd
+
+
+@pytest.fixture
+def dtype():
+    return pd.core.arrays.integer.Int8Dtype()
+
+
+@pytest.fixture
+def df(dtype):
+    return pd.DataFrame(
+        {
+            ("a", "x"): pd.Series(range(5), dtype=dtype),
+            ("b", "x"): pd.Series(range(5), dtype=float),
+        }
+    )
+
+
+def test_stack_preserves_extension_dtype(df, dtype):
```

Comment:
this can go in tests.frame.test_stack_unstack

Reply:
yes that's great, it indeed fits nicely under test_stack_nullable_dtype in tests.frame.test_stack_unstack.

Moved and refactored